### PR TITLE
IstioConfigDetails: notify user when resource changes outside Kiali

### PR DESCRIPTION
**Pull request by lucasponce**
Fixes kiali/kiali#1383

This PR introduces a poll mechanism to refresh the current editable resource with the aim to identify changes on the resourceVersion or errors (delete situation).

On this cases send a warning to the user, in the case when a resource is deleted, it disables "delete/save" actions only allowing to reload/cancel the page to move back to the list page.

Note that this PR uses a fixed timeout, as a first try, this also brings the scenario to enable "refresh" in list/details pages for similar use case: automatically updates.

But this last scenario will be discussed/implemented in a separate issue/PR.

### CODE REVIEW

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx [hunk 1]
```diff
@@ -136,10 +149,51 @@ class IstioConfigDetailsPage extends React.Component<RouteComponentProps<IstioCo
...
+            );
+            if (changed) {
+              AlertUtils.addWarning(
+                'Istio object type [' +
+                  type +
+                  '] name [' +
+                  props.object +
+                  '] has been modified externally to Kiali. New ResourceVersion:  ' +
+                  newResourceVersion
+              );
```
Comment (xeviknal) [inline src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx]: nitpick: The message is shown even if the user has another tab opened and changed the instance there. Meaning that the change has happened within Kiali console. 
Therefore, I'd put a wider message covering this scenario too. Something like 'sidecar:no-ip-tables has a newer version (9182178). Reload to see the new version.'

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx [hunk 2]
```diff
@@ -136,10 +149,51 @@ class IstioConfigDetailsPage extends React.Component<RouteComponentProps<IstioCo
...
+            }
+          }
+        })
+        .catch(error => {
+          this.setState({
+            isRemoved: true
+          });
+          AlertUtils.addError(
+            'Could not fetch Istio object type [' +
+              props.objectType +
```
Comment (xeviknal) [inline src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx]: Should we use the name naming as in the alerts for the validations `type:name`?

Screenshot of Kiali Console (23)

Screenshot of Kiali Console (24)

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx (hunk 1 above)
Comment (lucasponce) [reply src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx]: +1 true

File: src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx (hunk 2 above)
Comment (lucasponce) [reply src/pages/IstioConfigDetails/IstioConfigDetailsPage.tsx]: +1

**Pull request merged**